src/node/stage/approved.rs

```diff
@@ -558,12 +551,9 @@ impl Approved {
         let bounce_msg = Message::single_src(
             &core.full_id,
             DstLocation::Direct,
-            Variant::BouncedUnknownMessage {
-                message: msg_bytes,
-                parsec_version: self.consensus_engine.parsec_version(),
-            },
-            None,
+            Variant::BouncedUnknownMessage(msg_bytes),
             None,
+            Some(*self.shared_state.our_history.last_key()),
```

Comment:
Hm, I just realized that I might have misled you :flushed:  . We should not be using the `dst_key` field for this. The `dst_key` field is to indicate the latest key of the destination that we know. So if we are handling an unknown message from our own section, it would be OK to use `dst_key`, but this function is also used for handling messages from other sections and then it would make no sense to pass out own key. So I'm afraid it must be added back to `Variant::BouncedUnknownMessage`. Might call is `src_key` or something like that.

Alternatively, we can sort of abuse the `proof_chain` field for that, but that might be confusing, so not sure. Your call.

Reply:
I think a `src_key` field within the `Variant::BouncedUnknownMessage` will be more clear.